src/cli/localNode.ts

```diff
@@ -0,0 +1,98 @@
+import { ChildProcessWithoutNullStreams, spawn } from "child_process";
+import kill from "tree-kill";
+import { sleep } from "../utils/helpers";
+
+export class LocalNode {
+  readonly MAXIMUM_WAIT_TIME_SEC = 30;
+
+  readonly READINESS_ENDPOINT = "http://127.0.0.1:8070/";
+
+  process: ChildProcessWithoutNullStreams | null = null;
+
+  /**
+   * kills all the descendent processes
+   * of the node process, including the node process itself
+   */
+  stop() {
+    if (!this.process?.pid) return;
+    kill(this.process.pid);
+  }
+
+  /**
+   * Runs a local testnet and waits for process to be up.
+   *
+   * If local node process is already up it returns and does
+   * not start the process
+   */
+  async run() {
+    const nodeIsUp = await this.checkIfProcessIsUp();
+    if (nodeIsUp) {
+      return;
+    }
+    this.start();
+    await this.waitUntilProcessIsUp();
+  }
+
+  /**
+   * Starts the local testnet by running the aptos node run-local-testnet command
+   */
+  start() {
+    const cliCommand = "npx";
+    const cliArgs = ["aptos", "node", "run-local-testnet", "--force-restart", "--assume-yes", "--with-indexer-api"];
+
+    const childProcess = spawn(cliCommand, cliArgs);
+    this.process = childProcess;
+
+    childProcess.stderr?.on("data", (data: any) => {
+      const str = data.toString();
+      // Print local node output log
+      // eslint-disable-next-line no-console
+      console.log(str);
+    });
+
+    childProcess.stdout?.on("data", (data: any) => {
+      const str = data.toString();
+      // Print local node output log
+      // eslint-disable-next-line no-console
+      console.log(str);
+    });
+  }
+
+  /**
+   * Waits for the local testnet process to be up
+   *
+   * @returns Promise<boolean>
+   */
+  async waitUntilProcessIsUp(): Promise<boolean> {
+    let operational = await this.checkIfProcessIsUp();
+    const start = Date.now() / 1000;
+    let last = start;
+
+    while (!operational && start + this.MAXIMUM_WAIT_TIME_SEC > last) {
+      // eslint-disable-next-line no-await-in-loop
+      await sleep(1000);
+      // eslint-disable-next-line no-await-in-loop
+      operational = await this.checkIfProcessIsUp();
+      last = Date.now() / 1000;
+    }
+    return true;
```

Comment:
Should we return false, or error if it doesn't reach `operational`?  Or just return `operational` at the end.

Reply:
we can't return cause it would "solve" the promise - so we throw